integration-tests/src/lib.rs

```diff
@@ -128,22 +128,80 @@ where
     }
 }
 
-/// Initialize logging in tests.
-pub fn initialize_test_logging() {
+/// Initializes fallback logging.
+///
+/// This only exist to make sure all logs are always logged
+/// even if there is an accident and the global dispatch is
+/// used instead of the per-test dispatch.
+///
+/// There are a small number of logs where this is always the
+/// case (but we also normally don't care about) like when actix
+/// logs that it started a new worker thread.
+pub fn initialize_test_logging_fallback() {
     static ONCE: Once = Once::new();
     ONCE.call_once(|| {
-        logging::initialize(&logging::Config {
-            file: None,
-            level: LevelFilter::WARN,
-            // FIXME If we have json logging do fix the panic logging hook
-            //       to also log the backtrace instead of disabling the
-            //       panic hook.
-            install_panic_hook: false,
-        })
-        .unwrap();
+        let var = env::var_os("XAYN_TEST_FALLBACK_LOG");
+        let directives = var
+            .as_deref()
+            .map(|s| {
+                s.to_str()
+                    .expect("XAYN_TEST_FALLBACK_LOG must only contain utf-8")
+            })
+            .unwrap_or("warn");
+        tracing_subscriber::fmt()
+            .with_ansi(false)
+            .with_writer(TestWriter::default())
+            .with_env_filter(directives)
+            .init()
     });
```

Comment:
We no longer use the logging setup from the normal crate as we need something especially in the follow up PRs.

We might later (post the follow up PR) port this "new" setup back to the normal non-testing logging setup but that would be a braking change as it does things subtil differently.